src/tiger.lex

```diff
@@ -6,17 +6,28 @@ val linePos = ErrorMsg.linePos
 val commentDepth = ref 0;
 fun err(p1,p2) = ErrorMsg.error p1
 
-fun eof() = let val pos = hd(!linePos) in Tokens.EOF(pos,pos) end
+val currentLineStartPos = ref 1
+
+fun getLinePos yypos = (yypos - !currentLineStartPos)
+
+fun eof() =
+    let val pos = getLinePos(hd(!linePos)) in
+        let val eofToken = Tokens.EOF(pos,pos,!lineNum) in
+            (ErrorMsg.reset();
+                eofToken)
+        end
+    end
```

Comment:
Initially, this is what I did, but moved it to ErrorMsg because the position is already being tracked there. It felt off to have those two values separated